src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs

```diff
@@ -321,14 +321,25 @@ public bool TryCopyTo(Span<T> destination)
         /// For <see cref="ReadOnlySpan{Char}"/>, returns a new instance of string that represents the characters pointed to by the span.
         /// Otherwise, returns a <see cref="string"/> with the name of the type and the number of elements.
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Couldn't this achieve the same thing without this and without the extra method on span by just changing the if block to call the new method on string and making that method aggressive (or manually inlining it)?

Reply:
Part of this work is also avoiding passing `Span<T>` across our internal ABI surface. Many existing methods on `MemoryExtensions` and `Span` are marked with aggressive inlining, and their implementation is to deconstruct the span and to pass it to a worker method which takes `(ref T, int)`. This helps keep the individual components of the span enregistered without requiring the span instance itself to be laid out contiguously on the stack.

See also the "new codegen" sample in my original post, where `rsp` is never touched at all outside the standard method prolog / epilog.

Reply:
This method is allocating objects. I do not see why it needs to be micro-optimized like the MemoryExtension helpers.